editor/src/clj/editor/editor_extensions.clj

```diff
@@ -57,263 +99,165 @@
-
-(s/def :ext-module/get-commands lua-fn?)
-(s/def :ext-module/on-build-started lua-fn?)
-(s/def :ext-module/on-build-successful lua-fn?)
-(s/def :ext-module/on-build-failed lua-fn?)
-(s/def :ext-module/on-bundle-started lua-fn?)
-(s/def :ext-module/on-bundle-successful lua-fn?)
-(s/def :ext-module/on-bundle-failed lua-fn?)
-(s/def ::module
-  (s/keys :opt-un [:ext-module/get-commands
-                   :ext-module/on-build-started
-                   :ext-module/on-build-successful
-                   :ext-module/on-build-failed
-                   :ext-module/on-bundle-started
-                   :ext-module/on-bundle-successful
-                   :ext-module/on-bundle-failed]))
-
-(s/def :ext-action/action #{"set" "shell"})
-(s/def :ext-action/node-id ::node-id)
-(s/def :ext-action/property string?)
-(s/def :ext-action/value any?)
-(s/def :ext-action/command (s/coll-of string?))
-(defmulti action-spec :action)
-(defmethod action-spec "set" [_]
-  (s/keys :req-un [:ext-action/action
-                   :ext-action/node-id
-                   :ext-action/property
-                   :ext-action/value]))
-(defmethod action-spec "shell" [_]
-  (s/keys :req-un [:ext-action/action
-                   :ext-action/command]))
-(s/def ::action (s/multi-spec action-spec :action))
-(s/def ::actions (s/coll-of ::action))
-
-(s/def :ext-command/label string?)
-(s/def :ext-command/locations (s/coll-of #{"Edit" "View" "Assets" "Outline"}
-                                         :distinct true :min-count 1))
-(s/def :ext-command/type #{"resource" "outline"})
-(s/def :ext-command/cardinality #{"one" "many"})
-(s/def :ext-command/selection (s/keys :req-un [:ext-command/type :ext-command/cardinality]))
-(s/def :ext-command/query (s/keys :opt-un [:ext-command/selection]))
-(s/def :ext-command/active lua-fn?)
-(s/def :ext-command/run lua-fn?)
-(s/def ::command (s/keys :req-un [:ext-command/label
-                                  :ext-command/locations]
-                         :opt-un [:ext-command/query
-                                  :ext-command/active
-                                  :ext-command/run]))
-(s/def ::commands (s/coll-of ::command))
-(s/def :editor.editor-extensions.language-server/languages (s/coll-of string? :min-count 1 :distinct true))
-(s/def :editor.editor-extensions.language-server/command (s/coll-of string? :min-count 1))
-(s/def ::language-server
-  (s/keys :req-un [:editor.editor-extensions.language-server/languages
-                   :editor.editor-extensions.language-server/command]))
-(s/def ::language-servers (s/coll-of ::language-server))
-
-(defn- ext-state [project]
-  (-> project
-      (g/node-value :editor-extensions)
-      (g/user-data :state)))
+;; region script API
+
+(defn- mk-ext-get [project]
+  (rt/lua-fn [{:keys [rt evaluation-context]} lua-node-id-or-path lua-property]
+    (let [node-id-or-path (validation/ensure ::validation/node-id (rt/->clj rt lua-node-id-or-path))
```

Comment:
Hmm, IIRC the intention was to define a name "node-id" that, in the context of editor scripts, means "internal editor node id or proj-path", so that any place where editor returns or accepts this "node-id", both internal node id and proj-path are valid. Though I agree that the name is confusing. I'll rename the spec.